apps/docs/guides/building-codemods/codemod2.mdx

```diff
@@ -0,0 +1,177 @@
+
+```tsx
+// BEFORE 
+import {
+  AccessAlarm as AccessAlarmIcon,
+  ThreeDRotation as NiceIcon,
+} from '@mui/icons-material';
+
+//AFTER
+import AccessAlarmIcon from '@mui/icons-material/AccessAlarm';
+import NiceIcon from '@mui/icons-material/ThreeDRotation';
+
+```
+
+### Step 1: Initialize a new codemod.
+
+Run `npx codemod@latest init` in your desired location to store the codemod folder, and select `workflow` as your engine. This sets up the scaffolding for your codemod, making it easy to provide configuration and metadata. While optional for building and running a codemod, this step enhances discoverability and sharing.
+
+![Initializing codemod](/images/guides/codemod2/initializing.png)
+
+### Step 2: Find JS/TS files.
+
+From the previous step, we got a new directory `mui-named-import-to-direct-import` and file inside `src/index.ts`. Let's update this boilerplate file.
+
+We need to get all the files that potentially can have Material UI Icons imports
+
+```tsx
+import type { Api } from '@codemod.com/workflow';
+
+export async function workflow({ files }: Api) {
+  **await files()**
+}
+```
+
+Now let's get Javascript/Typescript context:
+
+```tsx
+import type { Api } from '@codemod.com/workflow';
+
+export async function workflow({ files }: Api) {
+  await files()
+	  **.jsFam()
+}**
+```
+
+### Step 3: Use ast-grep for pattern detection.
+
+Let's use [ast-grep](https://ast-grep.github.io/) which is built-in into our workflow engine
+
+```tsx
+import type { Api } from '@codemod.com/workflow';
+
+export async function workflow({ files }: Api) {
+  await files()
+	  .jsFam()
+	  .**astGrep("import { $$$_ } from '@mui/icons-material'"**)
+}
+```
+
+ast-grep allows matching code with a … code. `$$$_` pattern means any number of elements, in our case any number of named imports.
+
+### Step 4: Use AI for code transformation.
+
+Now that we have all the needed files and imports, we should fix it somehow. We can write some logic to handle different cases, like support of aliases or … we can just ask LLM to make changes
+
+```tsx
+import type { Api } from '@codemod.com/workflow';
+
+export async function workflow({ files }: Api) {
+  await files()
+    .jsFam()
+	  .**astGrep("import { $$$_ } from '@mui/icons-material'"**)
+    .**ai`
+      Migrate named imports from material ui icons to direct imports.
+      Make sure preserve import names.
+
+      Example before:
+      import { AccessAlarm as AccessAlarmIcon, ThreeDRotation } from '@mui/icons-material'
+
+      Example after:
+      import AccessAlarmIcon from '@mui/icons-material/AccessAlarm'
+      import ThreeDRotation from '@mui/icons-material/ThreeDRotation'
+    `;**
+	}
+```
+
+### Step 5: Run and test the codemod locally.
+
+Let's check it out:
+
+```bash
+npx codemod@latest mui-named-import-to-direct-import/src/index.ts \
+--OPENAI_API_KEY=$OPEN_API_KEY \
```

Comment:
is this --OPENAI_API_KEY hardcoded? 
or should we provide instructions to add that in codemodrc.json?
@arybitskiy